frame/contracts/src/exec.rs

```diff
@@ -1227,6 +1236,15 @@ where
 		});
 		Ok(())
 	}
+
+	fn reentrant_count(&mut self) -> u32 {
+		let id: AccountIdOf<Self::T> = self.top_frame().account_id.clone();
+		self.account_entrance_count(id) - 1u32
+	}
+
+	fn account_entrance_count(&mut self, account_id: AccountIdOf<Self::T>) -> u32 {
+		self.frames().filter_map(|f| Some(f.delegate_caller.is_none() && &f.account_id == &account_id)).count() as u32
```

Comment:
I guess you can do that in a cheaper way without filtering.

Reply:
i have no idea why filtering is considered not cheap enough